### PR TITLE
Add HiOp compatibility document

### PR DESCRIPTION
**Pull request by ashermancinelli**
As per offline discussion with @balos1 and @balay, we believe HiOp is ready for formal xSDK review.

### CODE REVIEW

**Comment by ashermancinelli** [comment]
HiOp depends on xSDK member MAGMA.

File: hiop-policy-compatibility.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+If you are not compatible, state what is lacking and what are your plans on how to achieve compliance.
+
+**Website:** https://github.com/LLNL/hiop
+
+### Mandatory Policies
+
+| Policy                 |Support| Notes                   |
+|------------------------|-------|-------------------------|
+|**M1.** Support portable installation through Spack. |Full| Fully-functional Spack package has been merged upstream. Our development team pushes updates continuously.|
+|**M2.** Provide a comprehensive test suite for correctness of installation verification. |Full| HiOp contains many test drivers for end-to-end integration tests as well as about 240 unit tests for each key linear algebra kernel.|
```
[the remark this answers is not in the document]
Comment (ashermancinelli) [reply hiop-policy-compatibility.md:16]: We do not support testing through spack yet.

[the remark this answers is not in the document]
Comment (ashermancinelli) [reply hiop-policy-compatibility.md:16]: Is this worth mentioning in the policy document, or were you just curious?

[the remark this answers is not in the document]
Comment (balos1) [reply hiop-policy-compatibility.md:16]: No need to mention it in the document, we were just gathering this information recently for an xSDK miletstone.

**Review by balos1 — APPROVED** [review]

**Pull request merged**